test/unit/Swaps.spec.ts

```diff
@@ -0,0 +1,48 @@
+import { expect } from 'chai';
+import Swaps, { SwapDeal } from '../../lib/swaps/Swaps';
+import Logger, { Level } from '../../lib/Logger';
+import { SwapPhase, SwapState, SwapRole } from '../../lib/types/enums';
+
+const loggers = Logger.createLoggers(Level.Warn);
+
+describe('Swaps', () => {
+  const takerDeal: SwapDeal = {
+    role: SwapRole.Taker,
+    phase: SwapPhase.SwapCreated,
+    state: SwapState.Active,
+    peerPubKey: '03029c6a4d80c91da9e40529ec41c93b17cc9d7956b59c7d8334b0318d4a86aef8',
+    orderId: 'f8a85c66-7e73-43cd-9ac4-176ff4cc28a8',
+    localOrderId: '1',
+    proposedQuantity: 0.01,
+    quantity: 0.01,
+    pairId: 'LTC/BTC',
+    price: 0.005,
+    takerCurrency: 'LTC',
+    makerCurrency: 'BTC',
+    takerAmount: 1000000,
+    makerAmount: 5000,
```

Comment:
I would like to see these values being directly tested, although their logic code is currently not testable. Might need to extract it to another static function.

Reply:
Yeah I don't know how I can sensibly break it down further. At the most basic level this is just multiplication. I can break out the `10000000` literal into a `static readonly` property on swaps, and can make a `quantity` variable and make these lines say `quantity * Swaps.SATOSHIS_PER_COIN` and `quantity * Swaps.SATOSHIS_PER_COIN * price`. Let me know if that sounds good to you or if you have a specific suggestion in mind.

Reply:
I was referring more to the assignment logic (Swaps.ts line 244), but we can also directly assert `calculateSwapAmounts` response with a few test cases.